### ISSUE TITLE
'ValueError: Invalid model' at '_calibration_wrapper.AddIntermediateTensors(model_content)'

### ISSUE THREAD

**Issue by kkimmk** [tensorflow/tensorflow#59464]
### 1. System information

- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Ubuntu 20.04
- TensorFlow installation (pip package or built from source): pip install 
- TensorFlow library (version, if pip package or github SHA, if built from source): 2.11

### 2. Description
When I try to convert diffusion model into TFLite with **Integer-only**, I got the following error.
```
Traceback (most recent call last):
  File "main.py", line 37, in <module>
    main()
  File "main.py", line 29, in main
    tflite_model = converter.convert()
  File "/home/minkyukim/projects/test/venv/lib/python3.8/site-packages/tensorflow/lite/python/lite.py", line 933, in wrapper
    return self._convert_and_export_metrics(convert_func, *args, **kwargs)
  File "/home/minkyukim/projects/test/venv/lib/python3.8/site-packages/tensorflow/lite/python/lite.py", line 911, in _convert_and_export_metrics
    result = convert_func(self, *args, **kwargs)
  File "/home/minkyukim/projects/test/venv/lib/python3.8/site-packages/tensorflow/lite/python/lite.py", line 1216, in convert
    return self._convert_from_saved_model(graph_def)
  File "/home/minkyukim/projects/test/venv/lib/python3.8/site-packages/tensorflow/lite/python/lite.py", line 1100, in _convert_from_saved_model
    return self._optimize_tflite_model(
  File "/home/minkyukim/projects/test/venv/lib/python3.8/site-packages/tensorflow/lite/python/convert_phase.py", line 215, in wrapper
    raise error from None  # Re-throws the exception.
  File "/home/minkyukim/projects/test/venv/lib/python3.8/site-packages/tensorflow/lite/python/convert_phase.py", line 205, in wrapper
    return func(*args, **kwargs)
  File "/home/minkyukim/projects/test/venv/lib/python3.8/site-packages/tensorflow/lite/python/lite.py", line 871, in _optimize_tflite_model
    model = self._quantize(model, q_in_type, q_out_type, q_activations_type,
  File "/home/minkyukim/projects/test/venv/lib/python3.8/site-packages/tensorflow/lite/python/lite.py", line 608, in _quantize
    result = _calibrator.add_intermediate_tensors(result)
  File "/home/minkyukim/projects/test/venv/lib/python3.8/site-packages/tensorflow/lite/python/optimize/calibrator.py", line 36, in add_intermediate_tensors
    return _calibration_wrapper.AddIntermediateTensors(model_content)
ValueError: Invalid model
```
If I modify the 17th line of `main.py` to `model = DiffusionModelV2Test(img_height=height, img_width=width, max_text_length=77)`, it generates TFLite output without error.
`DiffusionModelV2Test` is a model which I modified to make the model smaller.
The original model contains 3 downsamplings and 3 upsamplings, but `DiffusionModelV2Test` only contains 1 for each.
Also, when if I comment out from line 25 ~ 28, which makes the conversion to weight-only quantization, it generates TFLite without error as well.

So, My question is why does this error occurs?
And regarding the workaround with `DiffusionModelV2Test`, is there any limitation for the size of saved model when using calibration?

FYI.
The model I'm testing is diffusion model originally from keras-cv.
To make it convertable into TFLite, I only modified `keras.layers.Input` with `batch_size` specified as an argument.

### 3. Code

[reproducer gist](https://gist.github.com/kkimmk/239ffc431f9bea47513a3964f5f492ca)

**Comment by gaikwadrahul8** [comment]
Hi, @mohantym 

Could you please look into this issue ? Thank you!

**Comment by mohantym** [comment]
Hi @kkimmk !
Thanks for share your your observation on integer quantization of stablediffusion model. 


Could you check the [workaround ](https://github.com/tensorflow/tensorflow/issues/59259#issuecomment-1399214673)which suggests to use dynamic range int8 quantization . Please provide a colab gist to help saving the replication time . 

Thank you!

**Comment by kkimmk** [comment]
Hi @mohantym, Thank you for your suggestion!
First of all, I provided the reproducer gist at last of my first comment :)

I tested dynamic range int8 quantization without problem, but unfortunately what I'm trying to do is integer-only quantization.
Upon the workaround you've shared, which was very useful, I found out that TF model is first converted into TFLite model before applying quantization which caused the error due to the 2GB flatbuffer limitation. 
And therefore, I guess the only solution for now is to split diffusion model into 2 chunks and convert them separately.

Thank you for sharing the great workaround.
As I understood all the situations, now closing the issue.
